### DISCUSSION TITLE
Fix: Git for Windows Releases URL

### DISCUSSION THREAD

**Discussion by 16ajbm** [git-for-windows/git#3482] [Q&A]
Recently I have included Git for Windows in a Dockerfile. I was hoping to make the version manageable via a build argument, like `GIT_VERSION=2.33.0.2` and inject it into the URL to retrieve the installer `https://github.com/git-for-windows/git/releases/download/v${GIT_VERSION}windows/Git-${GIT_VERSION}-64-bit.exe`. The problem with the this is that the URL is actually something like `.../download/v2.33.0.windows.2/...` instead of `.../download/v2.33.0.2windows/...`. Is this intentional?  It seems strange, the installer follows the format `Git-2.33.0.2-64-bit.exe`. If this is intentional, any recommendations of how to deal with this in Docker? 

Thanks

**Comment by rimrul** [comment]
Yes, this is intentional. Our tags start with the tag naming system git/git uses, so `v2.33.0` or `v2.32.0-rc3` or similar. Our release is based on that upstream release.

We then attach a `.windows`, to indicate this is a Git for Windows tag and not a tag we inherited from git/git. And finally we add a `.1`, `.2`, .and so on to potentially create multiple releases based on one upstream release. The version number is basically the same thing with the `v` and the `.windows` removed, because most systems that expect version numbers don't handle text particularly well. 

You can probably achieve what you where trying to do using something like 

`https://github.com/git-for-windows/git/releases/download/v${GIT_VERSION%.*}.windows.${GIT_VERSION##*.}/Git-${GIT_VERSION}-64-bit.exe`

**Comment by dscho** [comment]
The best idea might be to use https://gitforwindows.org/latest-64-bit-installer.url. Unfortunately, GitHub Pages do not support redirects, therefore this is probably what you'll want to call:

```sh
curl -o Git-installer.exe $(curl -s https://gitforwindows.org/latest-64-bit-installer.url)
```
